### PR TITLE
Ai fix sort order

### PR DESCRIPTION
**Pull request by ron-murhammer**
Revert or fix a bunch of issues in #3323

### CODE REVIEW

**Comment by DanVanAtta** [comment]
@ron-murhammer even if you're confident in the fixes, I think we should still allow a review to happen:
1. communicates the changes to others as they review them
2. our policy is not to self-merge.

**Comment by ron-murhammer** [comment]
@DanVanAtta Only reason was this was a revert and AI only changes. It also made the AI essentially unplayable in the current pre-release.

File: game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java [hunk 1]
```diff
@@ -348,8 +347,13 @@ private static void findScrambleOptions(final PlayerID player, final Map<Territo
...
           if (maxCanScramble < canScrambleAir.size()) {
-            canScrambleAir.sort(Comparator.comparingDouble(
-                o -> ProBattleUtils.estimateStrength(to, Collections.singletonList(o), new ArrayList<>(), false)));
+            Collections.sort(canScrambleAir, (o1, o2) -> {
+              final double strength1 =
+                  ProBattleUtils.estimateStrength(to, Collections.singletonList(o1), new ArrayList<>(), false);
+              final double strength2 =
+                  ProBattleUtils.estimateStrength(to, Collections.singletonList(o2), new ArrayList<>(), false);
+              return Double.compare(strength2, strength1);
+            });
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java]: I'm curious why this couldn't be solved by adding a reversed call at the end.

Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/triplea/ai/pro/data/ProTerritoryManager.java]: @ron-murhammer #3382

**Pull request merged**